sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -936,14 +937,9 @@ export class ServiceBusMessage implements ReceivedMessage {
       return;
     }
     const receiver = this._context.getReceiver(this.delivery.link.name, this.sessionId);
+    throwIfMessageCannotBeSettled(receiver, DispositionType.complete, this.delivery.remote_settled);
 
-    if (receiver.receiveMode !== ReceiveMode.peekLock) {
```

Comment:
I don't see a throw for operation not supported error?

Reply:
Wouldn't be better to encapsulate the operation not supported in PeekLock mode as a different error ?
There could be other operations other than settlement.
Overall, `throwIfMessageCannotBeSettled` seems too generic and the checks and errors can be split up.

Reply:
Please see https://github.com/Azure/azure-sdk-for-js/blob/78cae5101a773526a96b1f59945fbafca0736cc5/sdk/servicebus/service-bus/src/util/errors.ts#L262

The errors are indeed different

Reply:
After offline discussions:

Good point on having a helper method for operations not supported in ReceiveAndDelete mode. It can be used for `renewMessageLock`, `receiveDeferredMessage` and `receiveDeferredMessages`.

https://github.com/Azure/azure-sdk-for-js/pull/2284/commits/b996f7d605644d2383ba34ddc65c0492cbe83d00 adds such a helper